python/paddle/v2/fluid/optimizer.py

```diff
@@ -190,6 +190,8 @@ def create_optimization_pass(self,
         # Create any accumulators
         program = loss.block.program
         with program_guard(program, startup_program):
+            global_block = framework.default_main_program().global_block()
+            start = len(global_block.ops)
```

Comment:
This approach assumes that the `optimizer.minimize` function can only be called once in the user's program definition.

Reply:
Yep, `optimizer.minimize` will append the optimize function for each parameter, for the currently Python API, `optimize.minimize` should be once.